src/components/shared/levelSelect.tsx

```diff
@@ -1,44 +1,70 @@
+
+import {useState, useEffect} from 'react';
 import '../../styles/levelSelect.scss';
-import {useState} from 'react';
 import Dropdown, { Option } from 'react-dropdown';
-import { BsArrowUpShort, BsArrowDownShort } from 'react-icons/bs';
-
-
-function LevelSelect(): JSX.Element {
+import { BsCaretLeftFill , BsCaretRightFill, BsFillCaretUpFill, BsFillCaretDownFill } from 'react-icons/bs';
+import { useHistory } from 'react-router-dom';
 
-  const pages = ['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+interface LevelSelectProps {
+  pages: string[];
+  locationPathName: string;
+}
 
-  const [currentPage, setCurrentPage] = useState(6); // change this to current page once set up
+function LevelSelect(props:LevelSelectProps): JSX.Element {
+  const { pages, locationPathName } = props;
 
   const createDropDownItems = ():Option[] => {
-    return pages.map((item) => {
+    return pages.map((item, index) => {
       return  ({
-        label: <div>{item}</div>,
+        label: <div>{index + 1}</div>,
         value: item,
       });
     });
   };
 
-  const pageArray = createDropDownItems();
+  const [currIndex, setcurrIndex] = useState(pages.indexOf(locationPathName));
+
+  const history = useHistory();
 
   const onPageChange = (currentOption:Option):void => {
-    // add functionality here to change the current page
-    // the selected option is passed in automatically
-    setCurrentPage(parseInt(currentOption.value));
+    const newIndex = pages.indexOf((currentOption.value));
+
+    if (newIndex == currIndex) {
+      /*
+ if the index wasn't actually changed, force component to rerender so
+ that "value" prop to dropdown still remains as the label on the dropdown
+*/
+      history.push(pages[currIndex]);
+    } else {
+      setcurrIndex(newIndex);
+    }
   };
```

Comment:
you can do `setIndex` instead of all this

Reply:
when I just call setIndex in onPageChange the dropdown doesn't keep what's passed in as "value" as what's displayed on it when the value of the current index is selected from the dropdown

Reply:
```
        onChange={(option) => setIndex(pages.indexOf(option.value))}
```